### PR TITLE
New room list basic styling and layout

### PR DESCRIPTION
**Pull request by turt2live**
~~Based on https://github.com/matrix-org/matrix-react-sdk/pull/4697~~ Merged
~~Based on https://github.com/matrix-org/matrix-react-sdk/pull/4687~~ Merged
For https://github.com/vector-im/riot-web/issues/13635

This is missing a large amount of CSS to make it pretty - it's largely just to get everything in the right position to be styled further. Some functionality is present, however.

Features:
* Rough positioning for components. Note how breadcrumbs and the filter box are aggressively against the style - this is because they are simply placed in and not styled here.
* Coarse features of the new top left menu. Missing obvious things like the menu button and other toggles.
* New possible style for badges (single dots), with support to switch to numbered badges. Badges are WIP.
* Selected room highlight (might become a box later, instead of a pill).
* Message preview placeholder - used for positioning and not updated here.
* List rendering minus all the buttons, hover states, etc. They exist for positioning.

*Note*: Design review is not required on this as it's just coarse positioning. Adjustments and further styling is slated for a later PR, if not an actual call where we move the pixels around a bit.

Screenshot:
image

With numbered badges:
image

### CODE REVIEW

**Comment by turt2live** [comment]
Flagging this up for review as it's structural

File: src/ActiveRoomObserver.js [hunk 1]
```diff
@@ -51,23 +55,23 @@ class ActiveRoomObserver {
         }
     }
 
-    _emit(roomId) {
```
Comment (t3chguy) [inline src/ActiveRoomObserver.js:54]: fun that ts/flow annotation works even in a non-flow js file

File: src/components/views/rooms/RoomList2.tsx [hunk 2]
```diff
@@ -200,6 +200,7 @@ export default class RoomList2 extends React.Component<IProps, IState> {
                     addRoomLabel={aesthetics.addRoomLabel}
                     isInvite={aesthetics.isInvite}
                     layout={this.state.layouts.get(orderedTagId)}
+                    showMessagePreviews={orderedTagId === DefaultTagID.DM}
```
Comment (t3chguy) [inline src/components/views/rooms/RoomList2.tsx:203]: Should this be the case?

image

Figma shows it on rooms too, wasn't it togglable per sublist?

Comment (turt2live) [reply src/components/views/rooms/RoomList2.tsx:203]: The feature is missing a setting and a whole lot of other work, which is why its a TODO and not implemented. Just needed something there to test the styling.

File: src/ActiveRoomObserver.js (hunk 1 above)
Comment (turt2live) [reply src/ActiveRoomObserver.js:54]: it's both great and awful.

File: src/components/views/rooms/RoomList2.tsx (hunk 2 above)
Comment (turt2live) [reply src/components/views/rooms/RoomList2.tsx:203]: also, for the running theme on these PRs: rubber stamped with *deferred*.

**Pull request merged**